### ISSUE TITLE
[bug] chains use okc to report an error

### ISSUE THREAD

**Issue by hhun** [WalletConnect/web3modal#1332]
### Link to minimal reproducible example

none

### Summary

`chains` use `okc` to report an error: 
> Uncaught Error: Could not find valid provider configuration for chain "OKC".
> You may need to add `jsonRpcProvider` to `configureChains` with the chain's RPC URLs.

`Chains` use `mainnet, bsc, arbitrum` and other functions without error.

``` js
<template>

  <w3m-network-button />
  <w3m-connect-button />
  <w3m-account-button />

  <button @click="modal.open()">Open Connect Modal</button>
  <button @click="modal.open({ view: 'Networks' })">Open Network Modal</button>

</template>

<script setup>
import { mainnet, okc } from '@wagmi/core/chains';
import { createWeb3Modal, defaultWagmiConfig, useWeb3Modal } from '@web3modal/wagmi/vue';

const chains = [mainnet, okc];
const projectId = 'YOUR_PROJECT_ID';
const wagmiConfig = defaultWagmiConfig({ chains, projectId, appName: 'Web3Modal' });

createWeb3Modal({ wagmiConfig, projectId, chains });

const modal = useWeb3Modal();
</script>
```

### List of related npm package versions

``` json
{
    "name": "Web3Modal-V3-Vue",
    "private": true,
    "version": "1.0.0",
    "type": "module",
    "scripts": {
        "dev": "vite",
        "build": "vite build",
        "preview": "vite preview"
    },
    "dependencies": {
        "vue": "^3.3.4",
        "@wagmi/core": "^1.4.1",
        "@web3modal/wagmi": "3.0.0-alpha.9",
        "viem": "^1.10.9"
    },
    "devDependencies": {
        "@vitejs/plugin-vue": "^4.2.3",
        "vite": "^4.4.5"
    }
}
```

**Comment by hhun** [comment]
> We will mention this more clearly in the documentation, but TL;DR is this: when you use `defaultWagmiConfig` helper you are using our pre-configured wagmi setup. You can see whats included by looking at advanced/custom sections of documentation https://docs.walletconnect.com/2.0/web3modal/v3/vue/about
> 
> By default provider (rpc) that we use is WalletConnect one and it doesn't support `okc`. So you will need to define your own wagmi configuration and add additional provider that supports that chain. Wagmi's documentation on provides should also be usefull.

But there is indeed okc in wagmi core chains.
但是 `@wagmi/core/chains` 里面确实有 `okc` 呢

**Comment by hhun** [comment]
``` js
const chains = [mainnet, bsc, okc]
const projectId = ENV.VITE_WALLET_CONNECT_PROJECT_ID

defaultWagmiConfig({ chains, projectId, appName: 'Web3Modal' })
createWeb3Modal({ wagmiConfig, chains, projectId })
```
Report an error
> runtime-core.esm-bundler.js:221 Uncaught Error: Could not find valid provider configuration for chain "OKC".
> You may need to add `jsonRpcProvider` to `configureChains` with the chain's RPC URLs.

Please tell me how to fix this problem

**Comment by 0xAsimetriq** [comment]
See "Custom" tab in this section
https://docs.walletconnect.com/2.0/web3modal/v3/react/about#implementation

You need to add a provider in addition to `walletConnectProvider` that supports okc. Usualy wagmi's `publicProvider` does the trick.

**Comment by hhun** [comment]
> See "Custom" tab in this section https://docs.walletconnect.com/2.0/web3modal/v3/react/about#implementation
> 
> You need to add a provider in addition to `walletConnectProvider` that supports okc. Usualy wagmi's `publicProvider` does the trick.

I looked at the custom tabs, but I still can’t write it.
Can you please help me write a demonstration code using okc as an example?

**Comment by 0xAsimetriq** [comment]
```tsx
import { createWeb3Modal } from '@web3modal/wagmi/react'
import { walletConnectProvider } from '@web3modal/wagmi'

// Add this <--
import { publicProvider } from 'wagmi/providers/public'

import { WagmiConfig, configureChains, createConfig } from 'wagmi'
import { mainnet } from 'wagmi/chains'
import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet'
import { InjectedConnector } from 'wagmi/connectors/injected'
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'

const projectId = 'YOUR_PROJECT_ID'

const { chains, publicClient } = configureChains([mainnet], [
    walletConnectProvider({ projectId }),
    // Add this <--
    publicProvider()
])

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: [
    new WalletConnectConnector({ options: { projectId, showQrModal: false } }),
    new InjectedConnector({ options: { shimDisconnect: true } }),
    new CoinbaseWalletConnector({ options: { appName: 'Web3Modal' } })
  ],
  publicClient
})


createWeb3Modal({ wagmiConfig, projectId, chains })

export default function App() {
  return (
    <WagmiConfig config={wagmiConfig}>

    </WagmiConfig>
  )
}
```

I think something like this will work, but if wagmi's publicProvider doesn't support okc, you'll need to make your own.

**Comment by hhun** [comment]
web3modal V3 **can't** run normally using `okc`
----
``` js
import { createWeb3Modal } from '@web3modal/wagmi/react'
import { walletConnectProvider } from '@web3modal/wagmi'

// Add this <--
import { publicProvider } from 'wagmi/providers/public'

import { WagmiConfig, configureChains, createConfig } from 'wagmi'
import { mainnet, bsc, okc } from 'wagmi/chains'
import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet'
import { InjectedConnector } from 'wagmi/connectors/injected'
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'

const projectId = 'YOUR_PROJECT_ID'

const { chains, publicClient } = configureChains([mainnet, bsc, okc], [
    walletConnectProvider({ projectId }),
    // Add this <--
    publicProvider()
])

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: [
    new WalletConnectConnector({ options: { projectId, showQrModal: false } }),
    new InjectedConnector({ options: { shimDisconnect: true } }),
    new CoinbaseWalletConnector({ options: { appName: 'Web3Modal' } })
  ],
  publicClient
})


createWeb3Modal({ wagmiConfig, projectId, chains })

function App() {

  return (
    <>
      <div>Vite + React</div>
      <WagmiConfig config={wagmiConfig} />
      <w3m-button />
    </>
  )
}

export default App
```
The above code will **not connect bsc and okc** when connecting to the wallet.
But after connecting, there will be bsc and okc in the switching network.

`Add this` specifically adds what code?
Could you please help me improve it?


web3modal V2 **can** run normally using `okc`
----
``` js
import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { configureChains, createConfig } from '@wagmi/core'
import { Web3Modal } from '@web3modal/html'
import { mainnet, bsc, okc } from '@wagmi/core/chains'

const chains = [mainnet, bsc, okc]
const projectId = 'YOUR_PROJECT_ID'

const { publicClient } = configureChains(chains, [w3mProvider({ projectId })])
const wagmiConfig = createConfig({
	autoConnect: true,
	connectors: w3mConnectors({ projectId, chains }),
	publicClient
})
const ethereumClient = new EthereumClient(wagmiConfig, chains)
const web3Modal = new Web3Modal(
	{
		projectId: projectId,
		themeMode: 'dark',
	},
	ethereumClient
)
```
V2 can run normally using `configureChains([mainnet, bsc, okc], [w3mProvider({ projectId })])`

**Comment by 0xAsimetriq** [comment]
@hhun try beta.2 with defaultWagmiConfig, should work

**Comment by hhun** [comment]
> @hhun try beta.2 with defaultWagmiConfig, should work

It can run normally using `v3.0.0-beta.2`
But, version v3 deletes the two parameters `chainImages` and `tokenImages`, and I hope to add these two parameters

**Comment by 0xAsimetriq** [comment]
Token images are available (please go over v3 docs) https://docs.walletconnect.com/2.0/web3modal/v3/about option for chain images will also be available before 3.0.0 stable release.
